### PR TITLE
Add Empty Java Client Project

### PR DESCRIPTION
**Pull request by mikaelm12**
Getting started on the Java Client work list.
Here: https://github.com/aspnet/SignalR/issues/2482

Let's just check something in to begin

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,11 @@ nuget.exe
 project.lock.json
 runtimes/
 .build/
+#Gradle
```
Comment (anurse) [inline .gitignore]: Spacing :). And put them at the bottom.

```
# Gradle files
...
```

File: clients/java/.settings/org.eclipse.buildship.core.prefs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+auto.sync=false
+build.scans.enabled=false
+connection.gradle.distribution=GRADLE_DISTRIBUTION(WRAPPER)
+connection.project.dir=
+eclipse.preferences.version=1
+gradle.user.home=
+offline.mode=false
+override.workspace.settings=true
```
Comment (anurse) [inline clients/java/.settings/org.eclipse.buildship.core.prefs]: I think we shouldn't check this file in. When I opened the project in VS Code it modified the file. Seems like it might be a per-user thing...

File: clients/java/build.gradle [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    api 'org.apache.commons:commons-math3:3.6.1'
+
+    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
+    implementation 'com.google.guava:guava:23.0'
+
+    // Use JUnit test framework
+    testImplementation 'junit:junit:4.12'
+
+    //Java JSON Library 
+    implementation 'com.google.code.gson:gson:2.8.5'
```
Comment (anurse) [inline clients/java/build.gradle]: Are these dependencies required to build or just sample dependencies that come with the template? It looks like the latter to me (`commons-math`, `guava`, `gson` don't seem to be used)

File: clients/java/.classpath [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (anurse) [inline clients/java/.classpath]: I think these can be auto-generated by a Gradle plugin: https://docs.gradle.org/current/userguide/eclipse_plugin.html

Maybe we should configure that and exclude this and the `.project` file from Git

File: clients/java/build.gradle (hunk 3 above)
Comment (mikaelm12) [reply clients/java/build.gradle]: I added this one and forgot to remove it. The rest came out of the box. Commented them out and it still builds successfully 🤷‍♂️

Comment (mikaelm12) [reply clients/java/build.gradle]: Except JUnit of course

**Comment by BrennanConroy** [comment]
Email Eilon about dependencies

File: clients/java/src/main/java/HubConnection.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (BrennanConroy) [inline clients/java/src/main/java/HubConnection.java]: `clients/java/src/main/java/filename.java` seems a bit much, maybe `clients/java/src/filename.java`?

File: clients/ts/package-lock.json [hunk 6]
```diff
@@ -1029,7 +1029,6 @@
       "requires": {
         "anymatch": "1.3.2",
         "async-each": "1.0.1",
-        "fsevents": "1.2.2",
```
Comment (BrennanConroy) [inline clients/ts/package-lock.json]: probably want to revert this change?

File: clients/java/src/main/java/HubConnection.java (hunk 5 above)
Comment (mikaelm12) [reply clients/java/src/main/java/HubConnection.java]: `src/main/java` is the standard convention for Java projects

**Comment by mikaelm12** [comment]
Opening a new PR
